partiql-system/src/main/java/org/partiql/system/PartiQLSession.java

```diff
@@ -0,0 +1,101 @@
+package org.partiql.system;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.partiql.spi.catalog.Catalogs;
+import org.partiql.spi.catalog.Namespace;
+import org.partiql.spi.catalog.Path;
+import org.partiql.spi.catalog.Session;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This is a package-private implementation of {@link Session}.
+ * @see PartiQLSessionBuilder
+ */
+public final class PartiQLSession implements Session {
```

Comment:
Session already has a builder and standard implementation, so I'm not sure we need to make these things public. You could internalize these by using the named builder pattern like `Session.builder()..build()` which has the PartiQL system installed, or use a `Session.bare()...build()` or something like that.